src/plugins/intel_cpu/src/nodes/reorder.cpp

```diff
@@ -175,77 +175,71 @@ void Reorder::prepareParams() {
             IE_THROW() << "Input memory didn't allocate.";
         if (getSelectedPrimitiveDescriptor() == nullptr)
             IE_THROW() << "Preferable primitive descriptor is not set.";
-
-        createReorderPrimitive(srcMemPtr->getDescWithType<DnnlMemoryDesc>()->getDnnlDesc(), srcMemPtr->getData(),
-                               dstMemPtr->getDescWithType<DnnlMemoryDesc>()->getDnnlDesc(), dstMemPtr->getData());
+        createReorderExecutor(srcMemPtr, dstMemPtr);
     }
 }
 
-void Reorder::createReorderPrimitive(const dnnl::memory::desc& srcDesc,
-                                     void* srcPtr,
-                                     const dnnl::memory::desc& dstDesc,
-                                     void* dstPtr) {
+struct ReorderExecutorKey {
+    dnnl::memory::desc src;
+    dnnl::memory::desc dst;
+    InferenceEngine::Precision src_prc;
+    InferenceEngine::Precision dst_prc;
+    size_t hash() const;
+    bool operator==(const ReorderExecutorKey& rhs) const;
+};
```

Comment:
We should think about the need to cache this executor. The thing is, the reorder primitives are already cached, additional converts initialization isn't costly. The only thing that may be amortized with caching is the attempts to instance the reorder primitive to test whether the precision combination is supported.
The most correct solution is storing the base `MemoryDesc` types in the key, of course this will make the hashing and comparison far more complicated, but it allows to describe an arbitrary reorder node, and switch from direct use of the reorder primitive to reorder executor.

Reply:
Agree, it is the most optimal solution. The cost is that hash computation will become complicated.
I also think maybe not cache executor will be a balance solution.